src/index.js

```diff
@@ -1,32 +1,46 @@
 import {defaultToToken, assertTokens} from './token';
 import {getTemplate, makeSubstitutions} from './template';
-import {calculateDelimiters} from './delimited';
+import {defaultSplitPlural, assertPluralForms} from './plurals';
 
 
 /**
  * Gettext as a tagged template string interpolator.
  *
  * For the interpolator, substitutions should be objects with a single key-value pair. The key
- * gives the `key` and `name` for the token and the `value` gives the value.
+ * gives the `key` and `name` for the token and the `value` gives the value. Substitutions with the
+ * same `name` must have `value` or an Error with result.
  *
  * Where all `substitutions` are `string` then a `string` is returned. Otherwise an `Array` is
  * returned. This allows React elements as substitutions. Elements must have an explicit `key` or
  * they will be assigned one automatically.
  *
- * Failure to pass a `translate` function will not result in an error, but no translation will
+ * Failure to pass a `gettext` function will not result in an error, but no translation will
  * occur.
  *
+ * Customisation of the `toToken` function is for advanced users only. Make reference to the source
+ * code.
+ *
+ * @throws Error On substitutions with duplicate `name`
  * @param {function} [gettext] Optional translation function, required for translation to occur
- * @param {function} [toToken] Optional custom token inference function, yeilds {name, key, value}
+ * @param {function} [toToken] Optional custom token inference function, yields {name, key, value}
+ * @param {string} [NODE_ENV] Reserved for testing
  * @returns {function(array, ...string):array|string} Template string interpolator
  */
 export const gettextFactory = ({
   gettext = x => x,
-  toToken = defaultToToken
+  toToken = defaultToToken,
+  NODE_ENV = (process.env || {}).NODE_ENV
 } = {}) => (strings, ...substitutions) => {
+  // get a msgid template with sensible token names
   const tokens = substitutions.map(toToken);
   const {names, values, msgid} = getTemplate(strings, tokens);
-  assertTokens(tokens, `Error in gettext(${msgid})`);
+
+  // validate unless production
+  if (NODE_ENV !== 'production') {
```

Comment:
I think you need to move this comment to line 32.

If you use the `DefinePlugin` properly it will replace `node.env`. On reflection I will not take the risk, given there is not an `EnvironmentPlugin` for Webpack2 and I do not know how that behaves.

So the default value on line 32 will become `process.env.NODE_ENV` as suggested.